app/src/main/java/io/github/droidkaigi/confsched2018/presentation/speaker/SpeakerDetailFragment.kt

```diff
@@ -87,11 +103,126 @@ class SpeakerDetailFragment : Fragment(), Injectable {
         }
     }
 
+    private fun initViewTransition(view: View, savedInstanceState: Bundle?) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return
+        }
+
+        ViewCompat.setTransitionName(
+                view.findViewById<View>(R.id.speaker_image),
+                arguments!!.getString(EXTRA_TRANSITION_NAME))
+
+        val transitionInflater = TransitionInflater.from(activity)
+
+        if (savedInstanceState == null) {
+            val enterTransition = transitionInflater.inflateTransition(R.transition.shared_element_arc)
```

Comment:
if we use `.apply { }`, then we don't need to keep an instance of the transition (I'm not sure if it makes it more readable or not though, because of the TransitionListener. what do you think?)
``` kotlin
activity?.window?.sharedElementEnterTransition = transitionInflater
        .inflateTransition(R.transition.shared_element_arc)
        .apply {
            duration = 400

            addListener(object : Transition.TransitionListener {
                override fun onTransitionEnd(p0: android.transition.Transition?) {
                    removeListener(this)
                    // No need to start reveal anim if user pressed back button during shared element transition
                    if (!isEnterTransitionCanceled) {
                        view.post(revealViewRunnable)
                    }
                }

                override fun onTransitionResume(p0: android.transition.Transition?) {
                }

                override fun onTransitionPause(p0: android.transition.Transition?) {
                    isEnterTransitionCanceled = true
                }

                override fun onTransitionCancel(p0: android.transition.Transition?) {
                    isEnterTransitionCanceled = true
                }

                override fun onTransitionStart(p0: android.transition.Transition?) {
                }
            })
        }
```